src/generic-container.test.ts

```diff
@@ -74,6 +75,25 @@ describe("GenericContainer", () => {
     await expect(fetch(url)).rejects.toThrowError();
   });
 
+  it("should build and start from a Dockerfile with build arguments", async () => {
+    const context = path.resolve(__dirname, "..", "docker-with-buildargs");
+    const container = await GenericContainer.fromDockerfile(withContext(context), withBuildArg("VERSION", "10-alpine"));
```

Comment:
Instead of exposing different ways to compose options, I think it'd be great if we could turn this into a builder. For example:

```
GenericContainer.fromDockerfile(context)
  ..withBuildArg(...)
  ..withX(...)
```

That way we don't have to expose different options and clients can use the type system to create their Dockerfile

Reply:
You can also use the type system with this config option. I understand you, but this way it was a smaller API change, than switching to a completely different pattern ... args vs builder.

But we can switch it to a builder pattern if you like it more ;)
I'll try to do this within the next 1-2 weeks.

Reply:
Okay. I moved everything to a builder pattern and added an example to the readme ;)